qiskit_ibm_runtime/noise_learner/noise_learner.py

```diff
@@ -0,0 +1,268 @@
+
+        # a circuit returning a two-qubit GHZ state
+        ghz = QuantumCircuit(2)
+        ghz.h(0)
+        ghz.cx(0, 1)
+
+        # another circuit returning a two-qubit GHZ state
+        another_ghz = QuantumCircuit(3)
+        another_ghz.h(0)
+        another_ghz.cx(0, 1)
+        another_ghz.cx(1, 2)
+        another_ghz.cx(0, 1)
+
+        pm = generate_preset_pass_manager(backend=backend, optimization_level=1)
+        circuits = pm.run([ghz, another_ghz])
+
+        # set the options
+        options = NoiseLearnerOptions()
+        options.layer_pair_depths = [0, 1, 10]
+
+        # run the noise learner job
+        learner = NoiseLearner(backend, options)
+        job = learner.run(circuits)
+
+    .. note::
+        Currently, the :class:`.NoiseLearner` is released an experimental feature.
+        As such, it is subject to change without notification and its stability is not
+        guaranteed.
+
+    Args:
+        mode: The execution mode used to make the primitive query. It can be:
+
+            * A :class:`Backend` if you are using job mode.
+            * A :class:`Session` if you are using session execution mode.
+            * A :class:`Batch` if you are using batch execution mode.
+
+            Refer to the `Qiskit Runtime documentation <https://docs.quantum.ibm.com/run>`_
+            for more information about the execution modes.
+
+        options: :class:`NoiseLearnerOptions`. Alternatively, :class:`EstimatorOptions` can be
+            provided for convenience, in which case the estimator options get reformatted into
+            noise learner options and all the irrelevant fields are ignored.
+
+    References:
+        1. E. van den Berg, Z. Minev, A. Kandala, K. Temme, *Probabilistic error
+           cancellation with sparse Pauli–Lindblad models on noisy quantum processors*,
+           Nature Physics volume 19, pages1116–1121 (2023).
+           `arXiv:2201.09866 [quant-ph] <https://arxiv.org/abs/2201.09866>`_
+
+    """
+
+    def __init__(
+        self,
+        mode: Optional[Union[BackendV2, Session, Batch]] = None,
+        options: Optional[Union[Dict, NoiseLearnerOptions, EstimatorOptions]] = None,
+    ):
+        self._mode, self._service, self._backend = _get_mode_service_backend(mode)
+        if isinstance(self._service, QiskitRuntimeLocalService):
+            raise ValueError("``NoiseLearner`` not currently supported in local mode.")
+
+        self._set_options(options)
+
+    @property
+    def options(self) -> NoiseLearnerOptions:
+        """The options in this noise learner."""
+        return self._options
+
+    def run(self, circuits: Iterable[Union[QuantumCircuit, EstimatorPubLike]]) -> RuntimeJobV2:
+        """Submit a request to the noise learner program.
+
+        This function breaks the given list of circuits into a list of unique layers, following
+        the strategy set by the ``twirling-strategy`` field specified in the ``options`` (see
```

Comment:
```suggestion
        the strategy set by the ``twirling.strategy`` field specified in the ``options`` (see
```

Reply:
I think it should be ``twirling-strategy`` because the ``options`` are NoiseLearnerOptions. In the description, they explain what will happen for the various strategies

Reply:
I see, I hadn't noticed the twirling prefix in the new options. I amend the suggestion to:

```suggestion
        the strategy set by the ``twirling_strategy`` field specified in the ``options`` (see
```

With the idea that anything in double backticks to always refer to some kind of code.